library/Freckle/App/OpenTelemetry/Context.hs

```diff
@@ -0,0 +1,116 @@
+module Freckle.App.OpenTelemetry.Context
+  ( HasHeaders (..)
+  , CustomTraceContext (..)
+  , extractContext
+  , injectContext
+  , processWithContext
+  ) where
+
+import Freckle.App.Prelude
+
+import Control.Error.Util (hush)
+import Control.Lens (Lens', lens, (&), (.~), (^.))
+import Control.Monad.Catch (MonadMask)
+import Data.Aeson (FromJSON, ToJSON)
+import Data.ByteString (ByteString)
+import Data.CaseInsensitive (CI)
+import qualified Data.CaseInsensitive as CI
+import Faktory.Job (Job, custom, jobOptions)
+import Faktory.Job.Custom (fromCustom, toCustom)
+import Faktory.JobOptions (JobOptions (..))
+import Freckle.App.OpenTelemetry
+  ( MonadTracer (..)
+  , SpanArguments
+  , inSpan
+  )
+import Freckle.App.OpenTelemetry.ThreadContext (withTraceContext)
+import Network.HTTP.Client (Request, requestHeaders)
+import Network.HTTP.Simple (setRequestHeaders)
+import Network.HTTP.Types.Header (Header)
+import OpenTelemetry.Context (Context)
+import OpenTelemetry.Context.ThreadLocal (attachContext, getContext)
+import OpenTelemetry.Propagator (Propagator, extract, inject)
+import OpenTelemetry.Trace.Core
+  ( getTracerProviderPropagators
+  , getTracerTracerProvider
+  )
+
+class HasHeaders a where
+  headersL :: Lens' a [Header]
```

Comment:
Feels a touch weird to be using `Network.HTTP.Types.Header` for things that may not be http. It's just a type alias, right? I might be less confused if we use the actual type: `(CI ByteString, ByteString)`